pkg/nar/util.go

```diff
@@ -7,3 +7,28 @@ import "strings"
 func IsValidNodeName(nodeName string) bool {
 	return !strings.Contains(nodeName, "/") && !strings.ContainsAny(nodeName, "\u0000")
 }
+
+// PathIsLexicographicallyOrdered checks if two paths are lexicographically ordered component by component.
+func PathIsLexicographicallyOrdered(path1 string, path2 string) bool {
+	if path1 > path2 {
+		path1Segments := strings.Split(path1, "/")
+		path2Segments := strings.Split(path2, "/")
+
+		// n is the lower number of segments of the two paths.
+		var n int
+		if len(path1Segments) < len(path2Segments) {
+			n = len(path1Segments)
+		} else {
+			n = len(path2Segments)
+		}
+
+		// check all segments individually
+		for i := 0; i < n; i++ {
+			if !(path1Segments[i] <= path2Segments[i]) {
+				return false
+			}
+		}
```

Comment:
If somebody wants to play with performance, a straight for loop over each character is probably more efficient. It would avoid the allocations and second pass over the string that strings.Split causes.

Reply:
Good catch! Implemented in https://github.com/nix-community/go-nix/pull/44.